pkg/dnsmasq/nanny.go

```diff
@@ -68,7 +70,26 @@ func (n *Nanny) Configure(args []string, config *config.Config) {
 	}
 
 	for domain, serverList := range config.StubDomains {
+		resolver := &net.Resolver{
+			PreferGo: true,
+			Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+				d := net.Dialer{}
+				return d.DialContext(ctx, "udp", kubednsServer)
+			},
+		}
 		for _, server := range serverList {
+			if isIP := (net.ParseIP(server) != nil); !isIP {
+				switch {
+				case strings.HasSuffix(server, "cluster.local"):
```

Comment:
Not necessarily in this PR, but we need to ensure this case is covered in e2e test: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/network/dns_configmap.go.

Reply:
@MrHohn Zihong, I will raise another PR in kubernetes repo to cover this case in e2e test